Source/WebCore/style/AnchorPositionEvaluator.h

```diff
@@ -37,21 +39,29 @@ namespace Style {
 
 class BuilderState;
 
-struct AnchorPositionedElementState {
+enum class AnchorPositionResolutionStage {
+    Initial,
+    HasCollectedCSSValues,
+    Resolved,
+};
+
+struct AnchorPositionedState {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    HashMap<String, Ref<Element>> anchorElements;
     HashSet<String> anchorNames;
-    bool finishedCollectingAnchorNames { false };
-    bool readyToBeResolved { false };
-    bool hasBeenResolved { false };
+    HashMap<String, Ref<Element>> anchorElements;
+    Vector<std::pair<CSSPropertyID, Ref<const CSSAnchorValue>>> insetPropertyCssValues;
+    Vector<std::pair<CSSPropertyID, std::optional<Length>>> insetPropertyResolvedValues;
+    AnchorPositionResolutionStage stage;
 };
 
-using AnchorPositionedStateMap = WeakHashMap<Element, std::unique_ptr<AnchorPositionedElementState>, WeakPtrImplWithEventTargetData>;
+using AnchorPositionedStates = WeakHashMap<Element, std::unique_ptr<AnchorPositionedState>, WeakPtrImplWithEventTargetData>;
+using EligibleAnchorsForName = HashMap<String, Vector<WeakRef<Element, WeakPtrImplWithEventTargetData>>>;
 
 class AnchorPositionEvaluator {
 public:
-    static Length resolveAnchorValue(const BuilderState&, const CSSAnchorValue&);
+    static Length getResolvedAnchorValue(const BuilderState&, const CSSAnchorValue&);
```

Comment:
nit: we don't use `getX` names per webkit code style. `resolvedAnchorValue` is probably OK.